strictdoc/backend/sdoc_source_code/reader_robot.py

```diff
@@ -0,0 +1,198 @@
+# mypy: disable-error-code="no-redef,no-untyped-call,no-untyped-def,type-arg,var-annotated"
+import sys
+from typing import List, Union
+
+from robot.api.parsing import (
+    Comment,
+    Documentation,
+    EmptyLine,
+    ModelVisitor,
+    Tags,
+    Token,
+    get_model,
+)
+
+from strictdoc.backend.sdoc.error_handling import StrictDocSemanticError
+from strictdoc.backend.sdoc_source_code.constants import FunctionAttribute
+from strictdoc.backend.sdoc_source_code.marker_parser import MarkerParser
+from strictdoc.backend.sdoc_source_code.models.function import Function
+from strictdoc.backend.sdoc_source_code.models.function_range_marker import (
+    FunctionRangeMarker,
+    RangeMarkerType,
+)
+from strictdoc.backend.sdoc_source_code.models.range_marker import (
+    LineMarker,
+    RangeMarker,
+)
+from strictdoc.backend.sdoc_source_code.models.source_file_info import (
+    SourceFileTraceabilityInfo,
+)
+from strictdoc.backend.sdoc_source_code.parse_context import ParseContext
+from strictdoc.backend.sdoc_source_code.processors.general_language_marker_processors import (
+    function_range_marker_processor,
+    line_marker_processor,
+    range_marker_processor,
+    source_file_traceability_info_processor,
+)
+from strictdoc.helpers.file_stats import SourceFileStats
+
+
+class SdocRelationVisitor(ModelVisitor):
+    """
+    Create functions from test cases in *.robot files. Create enriched markers.
```

Comment:
I cannot catch what is meant by `enriched` here. Let's remove it here or expand/elaborate.

Reply:
I meant "enriched" = "contains/associates with language aware information like the name of the test case or the line range of a test case". Dropped it.